tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -452,10 +452,17 @@ struct GreaterThan {
 // constraint mentioned above is hit. Tile size combinations lying outside this
 // frontier are either not possible, or are slower than the alternatives.
 //
-// It is instrumental to consider two subsets of the frontier:
-// - long side frontier: the union of the biggest tile size combinations for
+// It is instrumental to consider, for each data type, two subsets of the
+// corresponding frontier:
+// - long side frontier: the union of the biggest tile size combination for
 //   each legal long side len.
 // - non long side frontier: the frontier set minus the long side frontier.
+//
+// The following function TileSizePossibilityFrontierCheck will define the
+// frontier using only the long side frontier tile size combinations (since
+// one can easily extrapolate the entire frontier from this subset). It
```

Comment:
Since TileSizePossibilityFrontierCheck is nonrestrictive (meaning, the meaning of the sentence is the same if you leave it out), it should be set off with commas.

But probably a better way to write it would just be to say:

> TileSizePossibilityFrontierCheck defines the frontier ...